test/freenet/client/filter/OggPageTest.java

```diff
@@ -1,59 +1,57 @@
 package freenet.client.filter;
 
+import static freenet.client.filter.ResourceFileUtil.testResourceFile;
 import static org.junit.Assert.*;
 
 import java.io.ByteArrayOutputStream;
 import java.io.DataInputStream;
-import java.io.DataOutputStream;
 import java.io.IOException;
-import java.io.InputStream;
-import java.util.Arrays;
 
 import org.junit.Test;
 
 public class OggPageTest {
 	@Test
 	public void testStripNonsenseInterruption() throws IOException {
-		InputStream badData = getClass().getResourceAsStream("./ogg/nonsensical_interruption.ogg");
-		ByteArrayOutputStream filteredDataStream = new ByteArrayOutputStream();
-		DataOutputStream output = new DataOutputStream(filteredDataStream);
-		DataInputStream input = new DataInputStream(badData);
-		OggPage page = OggPage.readPage(input);
-		if(page.headerValid()) output.write(page.toArray());
-		page = OggPage.readPage(input);
-		if(page.headerValid()) output.write(page.toArray());
-		byte[] filteredData = filteredDataStream.toByteArray();
-		output.close();
-		input.close();
+		try (
+			ByteArrayOutputStream actualDataStream = new ByteArrayOutputStream();
+			ByteArrayOutputStream expectedDataStream = new ByteArrayOutputStream()
+		) {
+			testResourceFile(
+				"./ogg/nonsensical_interruption_filtered.ogg",
+				(input) -> readPages(expectedDataStream, input)
+			);
+			testResourceFile(
+				"./ogg/nonsensical_interruption.ogg",
+				(input) -> readPages(actualDataStream, input)
+			);
+			assertArrayEquals(expectedDataStream.toByteArray(), actualDataStream.toByteArray());
```

Comment:
Oh, wow, this test is bad. 😄 No need to do anything, though, this test was bad before the refactoring, too… 🙂